File: frontend/src/types/customer.ts
export type NextAction = '继续跟进' | '结束跟进'

export interface Customer {
  id: number
  name: string
  phone?: string
  address?: string
  notes?: string
  rate: number
  next_action: NextAction
  user_id: number
  track_count?: number
  last_track_at?: string
  created_at: string
  updated_at: string
  is_deleted: boolean
}

export interface CustomerWithLatestTrack {
  id: number
  name: string
  phone?: string
  address?: string
  rate: number
  notes?: string
  next_action: NextAction
  latest_track_time?: string
  latest_next_action?: NextAction
  latest_content?: string
  track_count: number
  user_id: number
  created_at: string
  updated_at: string
  is_deleted: boolean
}

export interface CustomerCreateRequest {
  name: string
  phone?: string | null
  address?: string | null
  notes?: string | null
  rate?: number
}

export interface CustomerUpdateRequest {
  name?: string
  phone?: string | null
  address?: string | null
  notes?: string | null
  rate?: number
}

export interface CustomerResponse {
  customer: Customer
}

export interface CustomerListResponse {
  customers: CustomerWithLatestTrack[]
  total: number
  page: number
  limit: number
}

export interface CustomerListQuery {
  page?: number
  limit?: number
  search?: string
}

File: backend/src/entities/customer.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};
use super::next_action::NextAction;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "customers")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub phone: Option<String>,
    pub email: Option<String>,
    pub company: Option<String>,
    pub address: Option<String>,
    pub notes: Option<String>,
    pub rate: i32,
    pub user_id: i32,
    pub created_at: ChronoDateTimeUtc,
    pub updated_at: ChronoDateTimeUtc,
    pub is_deleted: bool,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::user::Entity",
        from = "Column::UserId",
        to = "super::user::Column::Id"
    )]
    User,
    #[sea_orm(has_many = "super::customer_track::Entity")]
    CustomerTrack,
}

impl Related<super::user::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::User.def()
    }
}

impl Related<super::customer_track::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::CustomerTrack.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}

#[derive(Debug, Serialize, Deserialize)]
pub struct CustomerWithLatestTrack {
    pub id: i32,
    pub name: String,
    pub phone: Option<String>,
    pub rate: i32,
    pub notes: Option<String>,
    pub latest_track_time: Option<ChronoDateTimeUtc>,
    pub latest_next_action: Option<NextAction>,
    pub latest_content: Option<String>,
    pub created_at: ChronoDateTimeUtc,
}

#[derive(Debug, Deserialize)]
pub struct CreateCustomerRequest {
    pub name: String,
    pub phone: Option<String>,
    pub email: Option<String>,
    pub company: Option<String>,
    pub address: Option<String>,
    pub notes: Option<String>,
    pub rate: Option<i32>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateCustomerRequest {
    pub name: Option<String>,
    pub phone: Option<String>,
    pub email: Option<String>,
    pub company: Option<String>,
    pub address: Option<String>,
    pub notes: Option<String>,
    pub rate: Option<i32>,
}